x/wasm/types/authz.go

```diff
@@ -16,12 +18,92 @@ import (
 const gasDeserializationCostPerByte = uint64(1)
 
 var (
+	_ authztypes.Authorization         = &StoreCodeAuthorization{}
 	_ authztypes.Authorization         = &ContractExecutionAuthorization{}
 	_ authztypes.Authorization         = &ContractMigrationAuthorization{}
 	_ cdctypes.UnpackInterfacesMessage = &ContractExecutionAuthorization{}
 	_ cdctypes.UnpackInterfacesMessage = &ContractMigrationAuthorization{}
 )
 
+// NewStoreCodeAuthorization constructor
+func NewStoreCodeAuthorization(grants ...CodeGrant) *StoreCodeAuthorization {
+	return &StoreCodeAuthorization{
+		Grants: grants,
+	}
+}
+
+// MsgTypeURL implements Authorization.MsgTypeURL.
+func (a StoreCodeAuthorization) MsgTypeURL() string {
+	return sdk.MsgTypeURL(&MsgStoreCode{})
+}
+
+// Accept implements Authorization.Accept.
+func (a *StoreCodeAuthorization) Accept(ctx sdk.Context, msg sdk.Msg) (authztypes.AcceptResponse, error) {
+	var code []byte
+	var permission AccessConfig
+	switch msg := msg.(type) {
```

Comment:
pp: a switch can be very powerful but when used only for a single condition, it is not very readable to me. How about using an ordinary `if` as guard instead?
```go
	storeMsg, ok := msg.(*MsgStoreCode)
	if !ok{
		return authztypes.AcceptResponse{}, sdkerrors.ErrInvalidRequest.Wrap("unknown msg type")
	}
	code := storeMsg.WASMByteCode
	permission := *storeMsg.InstantiatePermission
```